WalletWasabi.Fluent/ViewModels/Wallets/WalletSettingsViewModel.cs

```diff
@@ -46,14 +48,24 @@ public WalletSettingsViewModel(WalletViewModelBase walletViewModelBase)
 				walletViewModelBase.RaisePropertyChanged(nameof(walletViewModelBase.PreferPsbtWorkflow));
 			});
 
-		this.WhenAnyValue(x => x.AutoCoinJoin)
-			.ObserveOn(RxApp.TaskpoolScheduler)
-			.Skip(1)
-			.Subscribe(x =>
+		SetAutoCoinJoin = ReactiveCommand.Create<bool>(async (x) =>
+		{
+			bool changed = false;
+			if (!_wallet.KeyManager.IsCoinjoinProfileSelected)
+			{
+				DialogResult<bool> isProfileSelected = await NavigateDialogAsync(new CoinJoinProfilesViewModel(_wallet.KeyManager, false), NavigationTarget.DialogScreen);
+				changed = isProfileSelected.Result;
+			}
+			if (changed)
```

Comment:
I do not get `changed` logic here. Isn't it `bool success` or `coinJoinProfileSelected`?

Reply:
I rewrote it and removed changed.